### PR TITLE
Correlations v3

### PR DESCRIPTION
**Pull request by damonge**
Correlations  module in v3

### CODE REVIEW

**Comment by nikfilippas** [comment]
Same here - as with `tracers_v3` and `bg_bolt_cl_v3`, this contains all the commit history from those, even though it is based on `v3_wip`. We should clean it up and only branch out from `v3_wip`, otherwise there is no point in having separate PRs for all these changes. That's the reason we are ditching the `star` branch (which passes the tests and contains almost all of the changes we are going to make).

File: pyccl/correlations.py [hunk 1]
```diff
@@ -220,28 +224,30 @@ def correlation_multipole(cosmo, a, beta, l, s, p_of_k_a=None):
...
     check(status, cosmo_in)
     if scalar:
         return xis[0]
     return xis
 
 
-def correlation_3dRsd(cosmo, a, s, mu, beta, use_spline=True, p_of_k_a=None):
+@warn_api(pairs=[('s', 'r')])
+def correlation_3dRsd(cosmo, *, a, r, mu, beta,
+                      use_spline=True, p_of_k_a=None):
```
Comment (nikfilippas) [inline pyccl/correlations.py:242]: Same here, let's swap to `r, a` to follow CCL unwritten rules.
Also, I think `p_of_k_a` should go before `use_spline`, as the latter is just a settings flag for the function.

**Review by nikfilippas — CHANGES_REQUESTED** [review]
Okay, I see you decided on being strict with KWO arguments after all, which I find good.
Some comments about the order of the KWO arguments (which don't affect how the code is run, just the order with which the args appear in the docs).

File: pyccl/correlations.py [hunk 2]
```diff
@@ -149,7 +151,8 @@ def correlation(cosmo, ell, C_ell, theta, type='NN', corr_type=None,
     return wth
 
 
-def correlation_3d(cosmo, a, r, p_of_k_a=None):
+@warn_api
+def correlation_3d(cosmo, *, a, r, p_of_k_a=None):
```
[the remark this answers is not in the document]
Comment (damonge) [reply pyccl/correlations.py:155]: Yes, sounds good.

File: pyccl/correlations.py [hunk 3]
```diff
@@ -190,15 +193,16 @@ def correlation_3d(cosmo, a, r, p_of_k_a=None):
     return xi
 
 
-def correlation_multipole(cosmo, a, beta, l, s, p_of_k_a=None):
+@warn_api(pairs=[('s', 'r'), ('l', 'ell')])
+def correlation_multipole(cosmo, *, a, beta, ell, r, p_of_k_a=None):
```
[the remark this answers is not in the document]
Comment (damonge) [reply pyccl/correlations.py]: Done

File: pyccl/correlations.py (hunk 1 above)
Comment (damonge) [reply pyccl/correlations.py]: Done

File: pyccl/correlations.py [hunk 4]
```diff
@@ -268,27 +274,28 @@ def correlation_3dRsd(cosmo, a, s, mu, beta, use_spline=True, p_of_k_a=None):
...
+                                            len(r), int(use_spline), status)
     check(status, cosmo_in)
     if scalar:
         return xis[0]
     return xis
 
 
-def correlation_3dRsd_avgmu(cosmo, a, s, beta, p_of_k_a=None):
+@warn_api(pairs=[('s', 'r')])
+def correlation_3dRsd_avgmu(cosmo, *, a, r, beta, p_of_k_a=None):
```
[the remark this answers is not in the document]
Comment (damonge) [reply pyccl/correlations.py]: done

File: pyccl/correlations.py (hunk 1 above)
Comment (nikfilippas) [reply pyccl/correlations.py]: Are we not moving `use_spline` after `p_of_k_a`?

**Review by nikfilippas — CHANGES_REQUESTED** [review]
Final thing.

File: pyccl/correlations.py [hunk 5]
```diff
@@ -311,36 +320,39 @@ def correlation_3dRsd_avgmu(cosmo, a, s, beta, p_of_k_a=None):
...
+        sigma (float or array-like): distance(s) times sine of the angle
+            (in Mpc).
+        a (float): scale factor.
         beta (float): growth rate divided by galaxy bias.
         p_of_k_a (:class:`~pyccl.pk2d.Pk2D`, `str` or None): 3D Power spectrum
             to integrate. If a string, it must correspond to one of the
             non-linear power spectra stored in `cosmo` (e.g.
             `'delta_matter:delta_matter'`). If `None`, the non-linear matter
             power spectrum stored in `cosmo` will be used.
+        use_spline (bool): whether to use a spline.
```
[the remark this answers is not in the document]
Comment (damonge) [reply pyccl/correlations.py]: done

**Comment by nikfilippas** [comment]
Why isn't coveralls reporting?

**Comment by damonge** [comment]
I see the report.

**Pull request merged**